src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -224,8 +214,33 @@ const JobHistory = _.flow(
               size: { basis: 150, grow: 0 },
               headerRenderer: () => h(HeaderCell, ['Status']),
               cellRenderer: ({ rowIndex }) => {
-                const { workflowStatuses, status } = submissions[rowIndex]
-                return h(Fragment, [statusCell(workflowStatuses), status === 'Aborting' && 'Aborting'])
+                const {
+                  methodConfigurationNamespace, methodConfigurationName, submissionId, workflowStatuses,
+                  status, submissionEntity
+                } = submissions[rowIndex]
+                return h(Fragment, [
+                  statusCell(workflowStatuses), status === 'Aborting' && 'Aborting',
+                  isTerminal(status) && workflowStatuses['Failed'] &&
+                  submissionEntity && submissionEntity.entityType.endsWith('_set') && h(TooltipTrigger, {
+                    content: 'Click to re-run'
```

Comment:
'Click to' is probably unnecessary, since it's clear that it's clickable. And this should probably say 'Re-run failures' since that's the exact action.